### ISSUE TITLE
Tags Data List autocomplete + Block Grid causing UI issues

### ISSUE THREAD

**Issue by AndyBoot** [leekelleher/umbraco-contentment#339]
### Which Contentment version are you using?

4.4.5

### Which Umbraco version are you using? For example: 8.14.1 - don't just write v8

12.0.1

### Bug summary

A property with Block Grid below a property with Contentment Data List using Tags editor causing the autocomplete to incorrectly appear. Appears to be a z-index issue.

### Steps to reproduce

![image](https://github.com/leekelleher/umbraco-contentment/assets/82056925/77ad7894-ff75-4a18-8e25-c4bfe5854cac)

Two fields:
- Categories - Contentment Data List - Tags
- Content - Block Grid

If the autocomplete height extends to the next property it causes autocomplete to appear beneath the Block Grid.

### Expected result / actual result

![image](https://github.com/leekelleher/umbraco-contentment/assets/82056925/d94590c4-d13a-4070-8dea-f33b79482774)

Manually adding a high z-index to the umb-property-editor <div> appears to resolve it.

Not sure if this is generally an Umbraco issue, or if Contentment can apply a z-index upon the autocomplete becoming active?

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

### What browsers are you seeing the problem on?

Chrome

**Comment by leekelleher** [comment]
Hi @AndyBoot, thanks for raising this.

I suspect this may be an Umbraco concern, as I've tried to reuse as much of the Tags UI as possible from Umbraco itself.  I'd expect if you tried to reproduce this scenario with Umbraco's own Tags property-editor, it'd have the same z-index issue.

As for the fix, I can't say with any certainty, but this CSS may need tweaking.
https://github.com/umbraco/Umbraco-CMS/blob/release-12.1.2/src/Umbraco.Web.UI.Client/src/less/typeahead.less#L4

**Comment by AndyBoot** [comment]
Hey @leekelleher, I figured this would be a general Umbraco UI issue rather than Contentment itself given the hacky z-index fix was within the containing element of the property rather than within the property. Thought I'd run it by yourself in the first instance just in case. 

Thanks for the suggestion. I'll take a look at the Umbraco CSS at some stage to see if I can help resolve it. For now though, I may just re-sort the properties by putting another property in between to create a gap to shroud the overarching issue haha

**Comment by leekelleher** [comment]
@AndyBoot In the meantime, on most of my Umbraco implementations I have a custom App_Plugin for any specific UI overrides/hacks, you could try adding something like this...

```css
.umb-tags-editor .tt-menu.tt-open {
    z-index: 1234 !important;
}
```
_(The `!important` is there because the `tt-menu` element has an inline style of `z-index: 100`.)_

I haven't tried that with your particular scenario, but some tinkering around with it should be able to make it workable.

**Comment by leekelleher** [comment]
I have added a workaround CSS fix in #341, for upcoming Contentment v4.5.0.

Closing this issue off. Updates and comments can be made on #341.
